src/rust/engine/fs/src/store.rs

```diff
@@ -1671,54 +1676,83 @@ mod remote {
     chunk_size_bytes: usize,
     upload_timeout: Duration,
     env: Arc<grpcio::Environment>,
-    channel: grpcio::Channel,
+    serverset: Serverset<grpcio::Channel>,
     authorization_header: Option<String>,
   }
 
   impl ByteStore {
+    #[cfg_attr(feature = "cargo-clippy", allow(too_many_arguments))]
     pub fn new(
-      cas_address: &str,
+      cas_addresses: &[String],
       instance_name: Option<String>,
-      root_ca_certs: Option<Vec<u8>>,
+      root_ca_certs: &Option<Vec<u8>>,
       oauth_bearer_token: Option<String>,
       thread_count: usize,
       chunk_size_bytes: usize,
       upload_timeout: Duration,
-    ) -> ByteStore {
+      backoff_config: BackoffConfig,
+    ) -> Result<ByteStore, String> {
       let env = Arc::new(grpcio::Environment::new(thread_count));
-      let channel = {
+
+      let channels = cas_addresses.iter().map(|cas_address| {
         let builder = grpcio::ChannelBuilder::new(env.clone());
-        if let Some(_root_ca_certs) = root_ca_certs {
+        if let Some(ref _root_ca_certs) = root_ca_certs {
           panic!("Sorry, we dropped secure grpc support until we can either make openssl link properly, or switch to tower");
-        /*
-          let creds = grpcio::ChannelCredentialsBuilder::new()
-            .root_cert(root_ca_certs)
-            .build();
-          builder.secure_connect(cas_address, creds)
-          */
+          /*
+            let creds = grpcio::ChannelCredentialsBuilder::new()
+              .root_cert(root_ca_certs)
+              .build();
+            builder.secure_connect(cas_address, creds)
+            */
         } else {
           builder.connect(cas_address)
         }
-      };
+      }).collect();
+
+      let serverset = Serverset::new(channels, backoff_config)?;
```

Comment:
I think that it might be a good idea (blocking, maybe) to take an initial value to loop with, and to then have callers randomize it (and keep it stable in tests of course). The goal would be to avoid causing a herd effect on the first server in your list on every startup.

Reply:
Done